pkg/controller/core/clusterqueue_controller.go

```diff
@@ -554,21 +554,16 @@ func (r *ClusterQueueReconciler) updateCqStatusIfChanged(
 }
 
 func (r *ClusterQueueReconciler) getWorkloadsStatus(cq *kueue.ClusterQueue) *kueue.ClusterQueuePendingWorkloadsStatus {
-	if cq.Status.PendingWorkloadsStatus == nil {
+	pendingWorkloads := r.qManager.GetSnapshot(cq.Name)
+	shouldUpdatePendingWorkloadStatus := cq.Status.PendingWorkloadsStatus == nil ||
```

Comment:
nit: extract to a function named like this. Then call the function from the same line as the `if`.

Reply:
done in https://github.com/kubernetes-sigs/kueue/pull/1069/commits/b5eb23383229431dbe25defb1c42b04687bb6503